doc/api-overview.md

```diff
@@ -0,0 +1,91 @@
+- Alternatively you can execute code on each item by using `Iter` or `IterAll`.
+- `Diff` provides an implementation to easily calculate the differences between two Noms lists
+
+### Noms Sets
+
+If you need to store and retrieve Noms values by their value instead of by their index then you could use a Noms set - [Go](https://godoc.org/github.com/attic-labs/noms/go/types#Set) and [JavaScript](https://docs.noms.io/59.0.2/#set). Since the storage and retrieval is based on the contents of the Noms values this eliminates duplicate values and the potential of having to scan Noms lists to determine if a value already exists in the collection.
+
+### Noms Maps
+
+Noms provides a data structure that is similar to a hash table - called `Map` - [Go](https://godoc.org/github.com/attic-labs/noms/go/types#Map) and [JavaScript](https://docs.noms.io/59.0.2/#map). Noms maps store keys and values. The keys and values in Noms maps are Noms values. Noms maps provide many similar methods to Noms lists and sets:
+
+- `Empty` can be used to test for an empty Noms map
+- `Len` returns the number of keys in the Noms map
+- `Has` can be used to test if a provided key is present in the Noms map
+- `Get` returns the Noms value for the provided key 
+- `MaybeGet` is like `Get` but also returns a boolean to indicate if a value exists for the provided key
+- `Set` stores the provided value for the given key
+- `Remove` deletes the provided key and value from the Noms map
+- `Diff` provides an implementation to easily calculate the differences between two Noms maps
+- `Iter`, `IterAll`, `IterFrom` all execute the provided function on the entries in the Noms map
+- `First` returns the first key/value in the Noms map
+- `Last` returns the last key/value pair in the Noms map
+
+### Noms Structs
+
+Noms structs provide a data structure to store and retrieve Noms values similar to Noms maps but the keys are strings and the values are Noms values. Noms structs provide type checking to ensure the keys being used and the values being set are as expected. 
+
+- `Get`
+- `MaybeGet`
+- `Set`
+- `Diff`
+
+*TODO: write about Noms structs here*
+
+## JavaScript Specifics
+
+### Flow, Promises, Unicorns, Oh My!
+
+Noms uses several JavaScript packages to improve source code quality, readability and performance.  You can see the dependencies by reviewing [package.json](https://github.com/attic-labs/noms/blob/master/js/noms/package.json). Many of these will not affect you, but a few definitely will. The usage of [Promises](https://www.promisejs.org/) and [Flow](https://flowtype.org/) type annotations will be noticeable in every JavaScript file and most APIs. If you are not already familiar with these projects or concepts it is worth your time to go learn the basics as it will make it easier for you to develop using the Noms JavaScript SDK.
```

Comment:
Better docs for Promise:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise